libethashseal/Ethash.cpp

```diff
@@ -42,6 +42,7 @@ Ethash::Ethash()
 	m_farm.setSealers(sealers);
 	m_farm.onSolutionFound([=](EthashProofOfWork::Solution const& sol)
 	{
+		Guard l(x_submit);
```

Comment:
I think we should also have this lock in `Ethash::generateSeal` where `m_sealing` is being changed.

Also, I don't like holding this lock while we call the callback `m_onSealGenerated`. Maybe let's try to limit the lock scope here?